domain/secret/service/watcher.go

```diff
@@ -46,9 +46,17 @@ func NewWatchableService(
 // WatchConsumedSecretsChanges watches secrets consumed by the specified unit
 // and returns a watcher which notifies of secret URIs that have had a new revision added.
 func (s *WatchableService) WatchConsumedSecretsChanges(ctx context.Context, unitName string) (watcher.StringsWatcher, error) {
-	ch := make(chan []string, 1)
-	ch <- []string{}
-	return watchertest.NewMockStringsWatcher(ch), nil
+	table, query := s.st.InitialWatchStatementForConsumedSecrets(unitName)
+	w, err := s.watcherFactory.NewNamespaceWatcher(
+		table, changestream.Update, query,
+	)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	processChanges := func(ctx context.Context, consumerIDs ...string) ([]string, error) {
+		return s.st.GetConsumedSecretURIs(ctx, unitName, consumerIDs...)
+	}
+	return newSecretChangeWatcher(w, s.logger, processChanges)
```

Comment:
If we can mutate the eventchange.Changed in the mapper, then we won't need this extra watcher anymore. But the question is should we allow to update the changed raw event content?